src/lua/init.c

```diff
@@ -798,6 +811,32 @@ luaopen_tarantool(lua_State *L)
 	return 1;
 }
 
+void
+tarantool_lua_load_modules(lua_State *L, const char * const *modules)
+{
+	lua_getfield(L, LUA_REGISTRYINDEX, "_LOADED");
+	for (const char * const *s = modules; *s; s += 2) {
+		const char *modname = *s;
+		const char *modsrc = *(s + 1);
+		const char *modfile = lua_pushfstring(L, "@builtin/%s.lua",
+						      modname);
+		if (luaL_loadbuffer(L, modsrc, strlen(modsrc), modfile))
+			panic("Error loading Lua module %s...: %s",
+			      modname, lua_tostring(L, -1));
+		lua_pushstring(L, modname);
+		lua_call(L, 1, 1);
+		if (!lua_isnil(L, -1)) {
+			/* package.loaded.modname = t */
+			lua_setfield(L, -3, modname);
+		} else {
+			lua_pop(L, 1); /* nil */
+		}
+		lua_pop(L, 1); /* chunkname */
+		builtin_modcache_put(modname, modsrc);
+	}
+	lua_pop(L, 1); /* _LOADED */
+}
```

Comment:
I do a similar thing in 4th commit in #8250, here we're clashing. I propose to wait a bit or rebase on top on my branch for a while.

Reply:
Reworked the PR based on #8250, please review (I don't see a way to set the merge into your fork branch, so PR have 8 commits now).